### PR TITLE
Add OGM harvest to CLI commands

### PR DESCRIPTION
**Pull request by ghukill**
### Purpose and background context

This PR fills out the stubbed OGM harvest command in click CLI commands, making the CLI ready when records normalization is complete for OGM harvests.

### How can a reviewer manually see the effects of these changes?

Set env vars:
```
WORKSPACE=dev
SENTRY_DSN=None
S3_RESTRICTED_CDN_ROOT=s3://cdn-origin-dev-222053980223/cdn/geo/restricted/
S3_PUBLIC_CDN_ROOT=s3://cdn-origin-dev-222053980223/cdn/geo/public/
GEOHARVESTER_SQS_TOPIC_NAME=geo-harvester-input-dev
```

It is now possible to run an OGM harvest via the CLI, that should produce some output indicating it has started:

```shell
pipenv run harvester --verbose harvest \
--harvest-type=incremental \
--from-date=2024-01-10 \
ogm
```

This will throw some errors, as the normalization logic is not complete, but it can be seen the harvest has started, with arguments like `--from-date` getting used as expected.

### Includes new or updated dependencies?
NO

### Changes expectations for external applications?
NO

### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/GDT-109

### Developer
- [ ] All new ENV is documented in README
- [ ] All new ENV has been added to staging and production environments
- [ ] All related Jira tickets are linked in commit message(s)
- [ ] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer(s)
- [ ] The commit message is clear and follows our guidelines (not just this PR message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The provided documentation is sufficient for understanding any new functionality introduced
- [ ] Any manual tests have been performed and verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

**Review by jonavellecuerdo — CHANGES_REQUESTED** [review]
Looks good but needs to be rebased with `main`! :)

**Comment by ghukill** [comment]
> Looks good but needs to be rebased with `main`! :)

@jonavellecuerdo - Rebased!  This was an interesting one... I'm pretty confident the code would have been okay without that step, but you're right, that doing so might result in a cleaner _shared_ git history.

File: tests/test_cli.py [hunk 1]
```diff
@@ -66,3 +66,40 @@ def test_cli_harvest_mit_full_legacy_single_success(
         obj={"START_TIME": perf_counter()},
     )
     assert result.exit_code == 0
+
+
+def test_cli_harvest_ogm_no_options_okay_success(runner, mocked_ogm_harvester):
```
Comment (ehanson8) [inline tests/test_cli.py:71]: Could this just be `test_cli_harvest_ogm_no_options_success` without the `okay`?

File: README.md [hunk 2]
```diff
@@ -136,4 +136,15 @@ Options:
...
 
-_Work in progress, OGM harvests not yet supported from CLI..._
\ No newline at end of file
+```text
+Usage: -c harvest ogm [OPTIONS]
+
+  Harvest and normalize OpenGeoMetadata (OGM) geospatial metadata records.
+
+Options:
+  --include-repositories TEXT  If set, limit to only these comma seperated
```
Comment (ehanson8) [inline README.md:145]: `limit to only *this* comma separated` instead?

**Review by ehanson8 — APPROVED** [review]
A few suggestions but code looks solid

File: README.md (hunk 2 above)
Comment (ehanson8) [reply README.md:145]: And same for `--exclude-repositories`

Comment (ghukill) [reply README.md]: Good call!  That language felt clunky.  Updated.

**Review by jonavellecuerdo — APPROVED** [review]

**Pull request merged**